huaweicloud/services/apig/resource_huaweicloud_apig_response.go

```diff
@@ -1,51 +1,81 @@
-	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+)
+
+type ErrorType string
+
+const (
+	ErrorTypeAuthFailure                 ErrorType = "AUTH_FAILURE"
+	ErrorTypeAuthHeaderMissing           ErrorType = "AUTH_HEADER_MISSING"
+	ErrorTypeAuthorizerFailure           ErrorType = "AUTHORIZER_FAILURE"
+	ErrorTypeAuthorizerConfigFailure     ErrorType = "AUTHORIZER_CONF_FAILURE"
+	ErrorTypeAuthorizerIdentitiesFailure ErrorType = "AUTHORIZER_IDENTITIES_FAILURE"
+	ErrorTypeBackendUnavailable          ErrorType = "BACKEND_UNAVAILABLE"
+	ErrorTypeBackendTimeout              ErrorType = "BACKEND_TIMEOUT"
+	ErrorTypeThrottled                   ErrorType = "THROTTLED"
+	ErrorTypeUnauthorized                ErrorType = "UNAUTHORIZED"
+	ErrorTypeAccessDenied                ErrorType = "ACCESS_DENIED"
+	ErrorTypeNotFound                    ErrorType = "NOT_FOUND"
+	ErrorTypeRequestParametersFailure    ErrorType = "REQUEST_PARAMETERS_FAILURE"
+	ErrorTypeDefault4XX                  ErrorType = "DEFAULT_4XX"
+	ErrorTypeDefault5XX                  ErrorType = "DEFAULT_5XX"
 )
 
 func ResourceApigResponseV2() *schema.Resource {
 	return &schema.Resource{
-		Create: resourceApigResponseV2Create,
-		Read:   resourceApigResponseV2Read,
-		Update: resourceApigResponseV2Update,
-		Delete: resourceApigResponseV2Delete,
+		CreateContext: resourceResponseCreate,
+		ReadContext:   resourceResponseRead,
+		UpdateContext: resourceResponseUpdate,
+		DeleteContext: resourceResponseDelete,
+
 		Importer: &schema.ResourceImporter{
-			State: resourceApigResponseResourceImportState,
+			StateContext: resourceCustomResponseImportState,
 		},
 
 		Schema: map[string]*schema.Schema{
 			"region": {
-				Type:     schema.TypeString,
-				Optional: true,
-				Computed: true,
-				ForceNew: true,
-			},
-			"group_id": {
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "The region where the API custom response is located.",
 			},
 			"instance_id": {
 				Type:     schema.TypeString,
 				Required: true,
 				ForceNew: true,
+				Description: "The ID of the dedicated instance to which the API group and the API custom response " +
+					"belongs.",
+			},
+			"group_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The ID of the API group to which the API custom response belongs.",
 			},
 			"name": {
 				Type:     schema.TypeString,
 				Required: true,
-				ValidateFunc: validation.StringMatch(
-					regexp.MustCompile("^[A-Za-z0-9_-]{1,64}$"), "The name consists of 1 to 64 characters. "+
+				ValidateFunc: validation.All(
+					validation.StringMatch(regexp.MustCompile(`^[\w-]*$`),
 						"Only letters, digits, hyphens(-), and underscores (_) are allowed."),
+					validation.StringLenBetween(1, 64),
+				),
```

Comment:
```suggestion
				ValidateFunc: validation.All(
					validation.StringMatch(regexp.MustCompile(`^[\w-]{1,64}$`),
						"Only letters, digits, hyphens(-), and underscores (_) are allowed."),
				),
```

It is more efficient to use one validation.

Reply:
The current writing method is better to prompt when customer make a mistake on format.